src/raw_reader.rs

```diff
@@ -7,13 +7,15 @@ use std::fmt::{Display, Formatter};
 /// `RawReader` is a shorthand for a [Reader](crate::Reader) implementation that returns [RawStreamItem]s and
 /// uses [RawSymbolToken] to represent its field names, annotations, and symbol values.
 pub trait RawReader: IonReader<Item = RawStreamItem, Symbol = RawSymbolToken> {
-    // Defines no additional functionality
+    // Mark the stream as complete. This allows the reader to understand when partial parses on
+    // data boundaries are not possible.
+    fn stream_done(&mut self);
```

Comment:
I think I prefer the phrasing in your comment. How about `stream_complete`?
```suggestion
    fn stream_complete(&mut self);
```